spring-ai-model/src/main/java/org/springframework/ai/template/TemplateRenderer.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2023-2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.ai.template;
+
+import java.util.Map;
+import java.util.function.BiFunction;
+
+/**
+ * Renders a template using a given strategy.
+ *
+ * @author Thomas Vitale
+ * @since 1.0.0
+ */
+public interface TemplateRenderer extends BiFunction<String, Map<String, Object>, String> {
```

Comment:
hard to know at this point, but wondering if this and the impls should go into spring-ai-commons as if there is a spring-ai-client-image in the future, it would also need templating support.

Reply:
ok, changes made. tx!